pystac_client/client.py

```diff
@@ -97,11 +116,6 @@ def open(
                 made to any part of this Catalog/API.
             parameters: Optional dictionary of query string parameters to
                 include in all requests.
-            ignore_conformance : Ignore any advertised Conformance Classes in this
-                Catalog/API. This means that
-                functions will skip checking conformance, and may throw an unknown
-                error if that feature is
-                not supported, rather than a :class:`NotImplementedError`.
```

Comment:
I'm a little cautious about removing this argument without warning downstream users well in advance. I propose that make this argument a noop, and warn w/ `FutureWarning` that it's going away in the next major release (v0.8 or v1).

Reply:
yes definitely! This PR needs to be tidied up for backwards compat.